my-app/src/Board.js

```diff
@@ -0,0 +1,88 @@
+import React, { useReducer } from 'react';
+import logo from './logo.svg';
+import Square from './Square';
+import {
+  BoardSquareContext,
+  BoardSetSquareContext,
+  SquareContext,
+} from './App.js';
+
+// main board component
+function Board() {
+  const boardSquares = React.useContext(BoardSquareContext);
+  const setBoardSquares = React.useContext(BoardSetSquareContext);
+
+  const initialSate = 'true';
+  const reducer = (state, action) => {
+    switch (action) {
+      case 'setXPlayer':
+        return !state;
+    }
+  };
+  const [isXPlayer, dispatch] = useReducer(reducer, initialSate);
+
+  // fucntion for when a sqaure is clicked
+  const whenClicked = index => {
+    // spread the items in boardSquare to new array to make a copy
+    const squares = [...boardSquares];
+    // check if square has a value
+    if (squares[index]) return;
+    // add next turn
+    squares[index] = isXPlayer ? 'X' : 'O';
+
+    // set the new state of the board squares
+    setBoardSquares(squares);
+
+    // change the state for the next turn
+    dispatch('setXPlayer');
+  };
+
+  let gameStatus;
+  const winner = findWinner(boardSquares);
+  gameStatus = winner
+    ? `Winner is ${winner}!`
+    : `It is${isXPlayer ? 'X' : 'O'}'s Turn...'`;
+
+  return (
+    <div className="board">
+      <h2 className={gameStatus}>{gameStatus}</h2>
+      <div className="boardRow">
+        {boardSquares.map((square, index) => (
+          <Square
+            id={index}
+            value={boardSquares[index]}
+            onClick={() => whenClicked(index)}
+          />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+// function to determine winner
+const findWinner = squares => {
```

Comment:
For functions like this, sometimes I'll put them in a `util` directory to keep things organized. I also think you could make a case that this should go on the `BoardContext`. One way to think about a context here is that it should contain all of the logic that you need to play a game of tic tac toe. In this case, that could be:

1. The state for the board
2. Whose turn it is
3. A way to take a turn
4. A way to figure out the winner

You have all four of those in your app, they are just spread out through the different components. If they were all on the context, each component could pull off what it needed when. 

Hopefully this comment makes sense - let me know if you have any follow-up questions!